v1.go

```diff
@@ -0,0 +1,16 @@
+package plugin
+
+// Plugin is the interface every plugin need to implement
+type Plugin interface {
```

Comment:
How about we add a `GetPluginInfo() Info`-method to the Plugin interface. In there, plugin authors can define some meta information for the plugin. This could be then displayed in the UI. Though I'm unsure if we need this.
```go
type Info struct {
    Version   string
    Author    string
    Website   string
    Name      string
    Description string
    License string
}
```

Reply:
Talking about metadata, actually, I was really unsure whether using file names to identify different plugins is a good idea. If a user reused a file name for a different plugin, the obsolete config data would be a big problem. Maybe we can adapt the concept of namespace to prevent all plugins from racing for the same global namespace. Eg. The implementation might look like this:
```golang
type PluginInfo struct {
    Version   string
    Author    string
    Name      string
    Website   string
    Description string
    License string
}
```
And plugins will be identified by `Author + "/" + Name` on initialization.


`GetPluginInfo()` should be a dedicated exported function in the plugin package rather than a receiver method to the `Plugin` interface, as it would be read before user context is loaded.

Reply:
True, it should be a dedicated function. We could add a gopackage string to the plugin info. Your netlify plugin would be `github.com/eternal-flame-AD/netlify-plugin` and we could use it as id in the plugin backend.